tests/unit/imports/server/generateJsonSchema.ts

```diff
@@ -0,0 +1,416 @@
+      assert.throws(() => collection.insert({ fixedLength: 'ab' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ pattern: '123' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ email: 'foo' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ uuid: 'foo' }), /Document failed validation/);
+
+      assert.doesNotThrow(() => collection.insert({ minLength: 'abcdef' }));
+      assert.doesNotThrow(() => collection.insert({ maxLength: 'ab' }));
+      assert.doesNotThrow(() => collection.insert({ fixedLength: 'abc' }));
+      assert.doesNotThrow(() => collection.insert({ pattern: 'abc' }));
+      assert.doesNotThrow(() => collection.insert({ email: 'jolly-roger@deathandmayhem.com' }));
+      assert.doesNotThrow(() => collection.insert({ uuid: 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11' }));
+    });
+
+    it('respects strings with multiple checks', async function () {
+      const schema = z.object({
+        _id: z.string(),
+        string: z.string().min(3).max(5).regex(/^[a-z]+$/),
+      });
+      const collection = await createTestCollection(schema);
+
+      assert.throws(() => collection.insert({ string: 'a' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ string: 'abcdef' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ string: '123' }), /Document failed validation/);
+
+      assert.doesNotThrow(() => collection.insert({ string: 'abcd' }));
+    });
+
+    it('allows strings with multiple regex checks', async function () {
+      const schema = z.object({
+        _id: z.string(),
+        string: z.string().regex(/^[a-z]+$/).regex(/^[a-z]{3}$/),
+      });
+      const collection = await createTestCollection(schema);
+
+      assert.throws(() => collection.insert({ string: 'a' }), /Document failed validation/);
+      assert.throws(() => collection.insert({ string: 'abcd' }), /Document failed validation/);
+
+      assert.doesNotThrow(() => collection.insert({ string: 'abc' }));
+    });
+
+    it('rejects unsupported checks', function () {
+      const schema = z.object({
+        _id: z.string(),
+        string: z.string().startsWith('a').optional(),
+      });
+      assert.throws(() => generateJsonSchema(schema), /Unsupported string check/);
+    });
+  });
+
+  describe('number fields', function () {
+    it('respects checks on numbers', async function () {
+      const schema = z.object({
+        _id: z.string(),
+        min: z.number().gt(3).optional(),
+        max: z.number().lt(3).optional(),
+        exclusiveMin: z.number().gte(3).optional(),
+        exclusiveMax: z.number().lte(3).optional(),
+        integer: z.number().int().optional(),
+        multipleOf: z.number().multipleOf(3).optional(),
+      });
+      const collection = await createTestCollection(schema);
+
+      assert.throws(() => collection.insert({ min: 3 }), /Document failed validation/);
```

Comment:
this is the epitome of nits, but I think I'd probably find the assertions here slightly more readable if they were grouped by behavior for each field rather than "accepted vs rejected", since I think it'd be easier to see the cutover points and the behavior at the boundary conditions.

Reply:
Quite reasonable 🙂. Fixed